### DISCUSSION TITLE
RFE: Filter for Backlinks pane

### DISCUSSION THREAD

**Discussion by mikro2nd** [zim-desktop-wiki/zim-desktop-wiki#2138] [Ideas]
A (very!) nice-to-have: A "Filter" for Backlinks. A pane at the top of the Backlinks pane with options to "include" or "exclude" the regex entered in a text-pane.

For example: I have some notes that are long-running, long-edited, and referred to by many, *many* other pages, including a *lot* of Journal pages. It would be really useful if I could *exclude* (suppress) all the (say) "Journal" pages from the backlinks view as a way to narrow the backlinks list down to just "more important/relevant" pages that refer to the current page. My UI concept would be a text-box that I could use to enter a regex (or prefix/substring) of page titles I want to include/exclude from the Backlinks list in view.

What think ye?

**Comment by jaap-karssenberg** [comment]
I like the idea :)

**Reply by mikro2nd** [reply]
If I can make some time I might poke a stick at it myself. (I keep having good intentions of getting more familiar with Python! Not much actual action so far, but maybe this will give me the compelling reason. After all, I expect it's merely a filter on a collection feeding a view-model or similar, right?)

**Comment by jaap-karssenberg** [comment]
Yes, the list is a view-model construct. There is a filtered model which
can be used for these kind of things. There is example code in the tasklist
plugin how to use that. Although the filter parsing in that plugin is a bit
complex, it does show how to set a filter boolean and make the view filter
based on that.

--- Jaap

**Reply by mikro2nd** [reply]
Upon further reflection, this "filter" idea would be a killer feature for the Link Map plugin, too. Mostly I find the Link Map useless precisely because it's so cluttered with Journal and Year/Month/Week-planner pages that the real idea-clusters are hopelessly obscured.

(And now I'm hijacking the thread with something that's really a different, albeit related, feature. Apologies.)

**Reply by introt** [reply]
@mikro2nd check out #1955

**Comment by jaap-karssenberg** [comment]
I think a live filter makes sense in both views. However if you always
exclude the journal namespace, you could also add a property with
include/exclude rules like in the task list plugin.

Jaap

Op vr 26 aug. 2022 14:05 schreef Mike ***@***.***>:

**Reply by mikro2nd** [reply]
I did think of a plugin config way to do it, but feel that a more dynamic/accessible way to filter will work better for most people most of the time. (Of course I might be wrong! ;))
